### DISCUSSION TITLE
Mounting and Saving Recordings to NFS Share

### DISCUSSION THREAD

**Discussion by albeedelacruz** [gtxaspec/wz_mini_hacks#107] [General]
Can anyone provide a mini/simple tutorial on how to mount a camera to an existing NFS share?

Thanks in advance

**Comment by gtxaspec** [comment]
`touch /opt/wz_mini/etc/configs/nfs.sh`

nfs.sh:
```
#!/bin/sh

until [ -d /media/mmc/record ]
do
	sleep 5
	echo "waiting for mmc to mount"
done
	echo "mmc mounted"
	mount -t nfs -o nolock,rw,noatime,nodiratime 192.168.1.1:/mnt/nfs/whatever /media/mmc/record &
exit
```

we place this in configs/ since it gets backed up during upgrades.  It waits for the camera to create the record directory, and then it mounts your share over it.

put the path `/opt/wz_mini/etc/configs/nfs.sh` into the custom script in your wz_mini.conf

**Reply by albeedelacruz** [reply]
Thanks I will try this.

**Reply by albeedelacruz** [reply]
Quick question. (not very Linux savvy but trying very hard), if my NFS share is behind a username/pw on my NAS and the path is 10.0.0.10/Backups/wyze will my mount command be:
`mount -t nfs -o nolock,rw,noatime,nodiratime un:pw@10.0.0.10:/Backups/Wyze /media/mmc/record &`

**Reply by gtxaspec** [reply]
I think it's `mount -t nfs -o nolock,rw,noatime,nodiratime, username=user,password=pass 10.0.0.10:/Backups/Wyze /media/mmc/record &`

**Reply by albeedelacruz** [reply]
I am getting this. Any idea?

`[root@WCV3:configs]# bash nfs.sh
mmc mounted
[root@WCV3:configs]# BusyBox v1.35.0 (2022-05-14 10:38:19 UTC) multi-call binary.

Usage: mount [OPTIONS] [-o OPT] DEVICE NODE

Mount a filesystem. Filesystem autodetection requires /proc.

        -a              Mount all filesystems in fstab
        -f              Dry run
        -i              Don't run mount helper
        -v              Verbose
        -r              Read-only mount
        -t FSTYPE[,...] Filesystem type(s)
        -T FILE         Read FILE instead of /etc/fstab
        -O OPT          Mount only filesystems with option OPT (-a only)
-o OPT:
        loop            Ignored (loop devices are autodetected)
        [a]sync         Writes are [a]synchronous
        [no]atime       Disable/enable updates to inode access times
        [no]diratime    Disable/enable atime updates to directories
        [no]relatime    Disable/enable atime updates relative to modification time
        [no]dev         (Dis)allow use of special device files
        [no]exec        (Dis)allow use of executable files
        [no]suid        (Dis)allow set-user-id-root programs
        [r]shared       Convert [recursively] to a shared subtree
        [r]slave        Convert [recursively] to a slave subtree
        [r]private      Convert [recursively] to a private subtree
        [un]bindable    Make mount point [un]able to be bind mounted
        [r]bind         Bind a file or directory [recursively] to another location
        move            Relocate an existing mount point
        remount         Remount a mounted filesystem, changing flags
        ro              Same as -r

There are filesystem-specific -o flags.`

**Reply by gtxaspec** [reply]
what version nfs is your server hosting?

**Reply by gtxaspec** [reply]
oops, sorry on mobile I added an extra space:

`mount -t nfs -o nolock,rw,noatime,nodiratime,username=user,password=pass 10.0.0.10:/Backups/Wyze /media/mmc/record &`

**Reply by albeedelacruz** [reply]
This is what I'm getting now.

`[root@WCV3:configs]# bash nfs.sh
mmc mounted
[root@WCV3:configs]# mount: mounting 10.0.0.11:/Surveillance on /media/mmc/record failed: Invalid argument`

I'm running NFSv4 on a SynologyNAS
I've tried 10.0.0.11:/volume1/Surveillance
and 10.0.0.11:/Surveillance/Wyze

 I get the same error.

**Reply by albeedelacruz** [reply]
My computer seems to connect fine to the NFS share without the UN/PW but i get a permission denied error when running it from the camera

![image](https://user-images.githubusercontent.com/36613296/174467029-aa1e1bd8-a20e-42df-ad12-c801a4d3bdf6.png)

**Reply by gtxaspec** [reply]
NFS is usually access restricted by IP, are your settings on the server set to allow the camera IP?

**Reply by albeedelacruz** [reply]
Yup I had the permissions set. Anyway I think I got it. Here's my nfs.sh and I'm no longer getting any errors. The camera is writing to the NAS now.

`#!/bin/sh

until [ -d /media/mmc/record ]
do
        sleep 5
        echo "waiting for mmc to mount"
done
        echo "mmc mounted"
        mount -t nfs -o nolock,rw,noatime,nodiratime 10.0.0.11:/volume1/Surveillance /media/mmc/record &
exit`

**Reply by gtxaspec** [reply]
great to hear!

**Reply by albeedelacruz** [reply]
One last question on this NFS recording feature (BTW, thanks for your help. IT IS WORKING GREAT): What happens if the disk gets full or the shared folder reaches its quota? Does wyze intelligently delete old files or do I manually need to do a cleanup on the drive?

**Reply by gtxaspec** [reply]
Not sure how the logic works, i've never run into that situation, but it would be the same as if your microsd was full.

**Reply by albeedelacruz** [reply]
That makes sense. Thanks!

**Reply by dcCMPY** [reply]
This was a great thread

After my v3 wouldnt auth or work with my Synology Surveillance Station, I resorted to this. 

My question, how does the recording work @albeedelacruz ? Is it one long coninuous file that is stored on your NFS Share? Or is this only for **Event Recording** ?

**Reply by albeedelacruz** [reply]
Its exactly the same as how it writes to the SD Card.
![image](https://user-images.githubusercontent.com/36613296/195984326-09189a85-785d-417d-9efa-2937dfdfbdf8.png)

You can extract the recording from the Wyze app itself but for me the most convenient way is to grab the files and use an MP4joiner app like Bandicut. If only there is a way to view the videos from Surveillance Station, that would be golden.

**Reply by dcCMPY** [reply]
Nice thanks

Were you ever able to add your cameras to Surveillance Station with RTSP enabled ?

I've tried but can never get Surveillance Station to authenticate against the camera

**Reply by Marcoske23** [reply]
> `touch /opt/wz_mini/etc/configs/nfs.sh`
> 
> nfs.sh:
> 
> ```
> #!/bin/sh
> 
> until [ -d /media/mmc/record ]
> do
> 	sleep 5
> 	echo "waiting for mmc to mount"
> done
> 	echo "mmc mounted"
> 	mount -t nfs -o nolock,rw,noatime,nodiratime 192.168.1.1:/mnt/nfs/whatever /media/mmc/record &
> exit
> ```
> 
> we place this in configs/ since it gets backed up during upgrades. It waits for the camera to create the record directory, and then it mounts your share over it.
> 
> put the path `/opt/wz_mini/etc/configs/nfs.sh` into the custom script in your wz_mini.conf

It works fine until you restart. At that point it stops saving. Do ihave to do the command on each reboot? any way to do automatically?

**Reply by albeedelacruz** [reply]
I have my custom scripts saved under \wz_mini\etc\rc.local.d now. Can you try?

**Comment by albeedelacruz** [comment]
Nope. I never had luck with RSTP since the beginning. I can get it to work but since I have lots of Wyze cams, running them all on Surveillance Station made it really really slow that it became unusable.On Oct 16, 2022, at 6:10 AM, dcCMPY ***@***.***> wrote:﻿
Nice thanks
Were you ever able to add your cameras to Surveillance Station with RTSP enabled ?

**Comment by dcCMPY** [comment]
Cool thanks for the reply

I can’t even get my single cam to connect!

The RTSP url works with VLC but that’s it.

Thanks,
Daniel
<https://aka.ms/o0ukef>
________________________________

**Comment by ericmagaha** [comment]
I am successfully able to SSH into the Wyze camera.  I can SCP files from it and SCP files to other hosts.

I am unable to get NFS working.  From the Wyze camera I get "Operation timed out".

`[root@WCV3:~]# mount -t nfs -o nolock,rw,noatime,nodiratime 192.168.1.200:/home/eric/wyze/camera1 /media/mmc/record
mount: mounting 192.168.1.200:/home/eric/wyze/camera1 on /media/mmc/record failed: Operation timed out`

I can successfully mount it from another linux box on my network fine.

`sudo mount -t nfs 192.168.1.200:/home/eric/wyze/camera1 foo1`

I even copied the nfs.sh as indicated earlier in this thread and updated the custom script path in wz_mini.conf.  It is not mounting.

Any ideas?

Thanks in advance.

**Comment by albeedelacruz** [comment]
Nfs permissions maybe?Sent from my iPhoneOn Nov 5, 2022, at 10:01 AM, ericmagaha ***@***.***> wrote:﻿
I am successfully able to SSH into the Wyze camera.  I can SCP files from it and SCP files to other hosts.
I am unable to get NFS working.  From the Wyze camera I get "Operation timed out".
***@***.***:~]# mount -t nfs -o nolock,rw,noatime,nodiratime 192.168.1.200:/home/eric/wyze/camera1 /media/mmc/record mount: mounting 192.168.1.200:/home/eric/wyze/camera1 on /media/mmc/record failed: Operation timed out
I can successfully mount it from another linux box on my network fine.
sudo mount -t nfs 192.168.1.200:/home/eric/wyze/camera1 foo1
I even copied the nfs.sh as indicated earlier in this thread and updated the custom script path in wz_mini.conf.  It is not mounting.
Any ideas?
Thanks in advance.

**Comment by ericmagaha** [comment]
Got it working by adding the NFS version to the command.
`mount -t nfs -o nfsvers=4,nolock,rw,noatime,nodiratime 192.168.1.200:/home/eric/wyze/camera1 foo1`

This works if `ENABLE_NFSv4` is set to false or true in wz_mini.conf.  Setting ENABLE_NFSv4 to true did not work by itself.

End result, `ENABLE_NFSv4="false"` in my config file and I am using `nfsvers=4` in the mount command.
